### PR TITLE
[VEG-1389] Upgrade outdated packages

### PR DESCRIPTION
**Pull request by txu-gsd**
@zendesk/vegemite 

## Description

Upgrade the outdated packages

## Detail

The following packages have been upgraded:
- oclif/dev-cli -> oclif/core
- lerna 3.20.2 -> 5.1.8
- lerna-change-log 1.0.0 -> 2.2.0
- mocha 5 -> 10
- nyc 14 -> 15.1.0
- node-fetch -> axios
- ts-node 10.8.1 -> 10.9.1

The following commands have ben ran with success
- test
- test:functional
- lint
- dev
- link:bin
- type:check
- postinstall
- audit:fix

## Checklist

- [N/A ] :guardsman: includes new unit and functional tests

### CODE REVIEW

**Comment by Frederick888** [comment]
```
$ yarn audit
yarn audit v1.22.19
0 vulnerabilities found - Packages audited: 1147
✨  Done in 1.20s.

$ yarn outdated
yarn outdated v1.22.19
info Color legend :
 "<red>"    : Major Update backward-incompatible updates
 "<yellow>" : Minor Update backward-compatible features
 "<green>"  : Patch Update backward-compatible bug fixes
Package     Current  Wanted   Latest Workspace          Package Type    URL
@types/node 14.18.22 14.18.22 18.0.6 zcli-monorepo      devDependencies https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/node
chalk       4.1.2    4.1.2    5.0.1  @zendesk/zcli-apps dependencies    https://github.com/chalk/chalk#readme
chalk       4.1.2    4.1.2    5.0.1  @zendesk/zcli-core dependencies    https://github.com/chalk/chalk#readme
chalk       4.1.2    4.1.2    5.0.1  @zendesk/zcli      dependencies    https://github.com/chalk/chalk#readme
✨  Done in 4.33s.
```

File: packages/zcli-apps/src/commands/apps/server.ts [hunk 1]
```diff
@@ -37,10 +37,10 @@ export default class Server extends Command {
   static strict = false
 
   async run () {
-    const { flags } = this.parse(Server)
+    const { flags } = await this.parse(Server)
```
Comment (token-cjg) [inline packages/zcli-apps/src/commands/apps/server.ts:40]: Any reason for the extra `await`s? Not a critique, just out of curiosity

File: packages/zcli-apps/src/lib/package.ts [hunk 2]
```diff
@@ -47,11 +47,11 @@ export const validatePkg = async (pkgPath: string) => {
   form.append('file', fs.createReadStream(pkgPath))
   const res = await request.requestAPI('api/v2/apps/validate', {
     method: 'POST',
-    body: form
+    data: form
   })
 
   if (res.status !== 200) {
-    const { description } = await res.json()
+    const { description } = await res.data
```
Comment (token-cjg) [inline packages/zcli-apps/src/lib/package.ts:54]: 👍

File: packages/zcli-apps/src/utils/createApp.ts [hunk 3]
```diff
@@ -14,20 +14,20 @@ export const uploadAppPkg = async (pkgPath: string): Promise<any> => {
   const pkgBuffer = fs.createReadStream(pkgPath)
   formData.append('uploaded_data', pkgBuffer)
   const response = await request.requestAPI('api/v2/apps/uploads.json', {
-    body: formData,
+    data: formData,
```
Comment (token-cjg) [inline packages/zcli-apps/src/utils/createApp.ts:17]: 👍

File: packages/zcli-apps/src/utils/createApp.ts [hunk 4]
```diff
@@ -41,20 +41,20 @@ export const deployApp = async (method: string, url: string, upload_id: number,
     installationPayload = { upload_id }
   }
   const installationOptions = {
-    body: JSON.stringify(installationPayload),
-    method,
+    data: JSON.stringify(installationPayload),
+    method: method,
```
Comment (token-cjg) [inline packages/zcli-apps/src/utils/createApp.ts:45]: any reason for being explicit here?

File: packages/zcli-apps/src/commands/apps/server.ts (hunk 1 above)
Comment (Frederick888) [reply packages/zcli-apps/src/commands/apps/server.ts:40]: Previous:
image

Now it's a promise:
image

File: packages/zcli-apps/tests/functional/server.test.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import { expect, test } from '@oclif/test'
 import * as path from 'path'
 import * as http from 'http'
-import fetch from 'node-fetch'
+import axios from 'axios'
```
Comment (token-cjg) [inline packages/zcli-apps/tests/functional/server.test.ts:4]: 👍

File: packages/zcli-apps/src/utils/createApp.ts (hunk 4 above)
Comment (Frederick888) [reply packages/zcli-apps/src/utils/createApp.ts]: Fixed in cd0973df173afd6322f338021f3825bb0ff97efb

File: packages/zcli-core/src/lib/auth.ts [hunk 6]
```diff
@@ -1,8 +1,8 @@
 import { CLIError } from '@oclif/errors'
 import * as chalk from 'chalk'
-import cli from 'cli-ux'
+import { CliUx } from '@oclif/core'
 import Config from './config'
-import fetch from 'node-fetch'
+import axios from 'axios'
```
Comment (token-cjg) [inline packages/zcli-core/src/lib/auth.ts:5]: 👍

File: packages/zcli/bin/run [hunk 7]
```diff
@@ -1,5 +1,5 @@
-#!/usr/bin/env node
+#!/usr/bin/env ts-node
```
Comment (token-cjg) [inline packages/zcli/bin/run:1]: 👍

**Review by token-cjg — APPROVED** [review]

File: packages/zcli/src/commands/profiles/list.ts [hunk 8]
```diff
@@ -25,7 +24,7 @@ export default class List extends Command {
         }
       }
     }, {
-      printLine: this.log
+      printLine: this.log.bind(this)
```
Comment (etanxing) [inline packages/zcli/src/commands/profiles/list.ts:27]: Without `.bind(this)`, does it cause any error? It's similar to https://github.com/oclif/core/blob/main/src/cli-ux/README.md#cliuxuxtable
```
import {Command, CliUx} from '@oclif/core'
import axios from 'axios'

export default class Users extends Command {
  static flags = {
    ...CliUx.ux.table.flags()
  }

  async run() {
    const {flags} = this.parse(Users)
    const {data: users} = await axios.get('https://jsonplaceholder.typicode.com/users')

    CliUx.ux.table(users, {
      name: {
        minWidth: 7,
      },
      company: {
        get: row => row.company && row.company.name
      },
      id: {
        header: 'ID',
        extended: true
      }
    }, {
      printLine: this.log,
      ...flags, // parsed flags
    })
  }
}
```

Comment (Frederick888) [reply packages/zcli/src/commands/profiles/list.ts:27]: Yes

```
tree 7095bce4f1ecfed073df5052eb0b00d7e644f5d3
parent 66c2e8ddb06c43ab6ca3fcd3cf9bf39a490b3c58
author Frederick Zhang <frederick.zhang@zendesk.com> Wed Jul 20 11:45:21 2022 +1000
committer Frederick Zhang <frederick.zhang@zendesk.com> Wed Jul 20 16:37:56 2022 +1000

Fix profiles:list TypeError this.jsonEnabled

See [1].

[1] https://github.com/oclif/core/issues/377

diff --git a/packages/zcli/src/commands/profiles/list.ts b/packages/zcli/src/commands/profiles/list.ts
index 2facb66..2539cbf 100644
--- a/packages/zcli/src/commands/profiles/list.ts
+++ b/packages/zcli/src/commands/profiles/list.ts
@@ -23,9 +23,9 @@ export default class List extends Command {
           return log
         }
       }
     }, {
-      printLine: this.log
+      printLine: this.log.bind(this)
     })
   }
 
   async run () {
```

Comment (etanxing) [reply packages/zcli/src/commands/profiles/list.ts:27]: ok I found https://github.com/oclif/core/issues/377 so their example is wrong 🙀

Comment (Frederick888) [reply packages/zcli/src/commands/profiles/list.ts:27]: Why read documentation when you can try a thousand things and waste 5 hours on it /s 🤪

**Review by etanxing — APPROVED** [review]

**Comment by txu-gsd** [comment]
Windows checks for `yarn dev` still having that module issue 
image

**Review by cl8au — APPROVED** [review]

File: .mocharc.json [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+    "package": "./package.json",
+    "reporter": "spec",
+    "slow": "75",
+    "timeout": "5000",
+    "ui": "bdd",
+    "watch-files": ["lib/**/*.js", "test/**/*.js"],
+    "watch-ignore": ["lib/vendor"],
+    "require": ["ts-node/register"],
+    "recursive": true
+}
```
Comment (cl8au) [inline .mocharc.json:13]: new line? :)

Comment (Frederick888) [reply .mocharc.json]: Addressed by d8831aa5a7691d1ff9dfa7b4309b851d0ff384d8

**Review by cl8au — APPROVED** [review]

**Pull request merged**